ado/ipacheckenum.ado

```diff
@@ -145,29 +160,50 @@ program ipacheckenum
 	
 	/* loop through the duration variables, calculate the mean
 	   and update the duration sheet */
-	foreach var of varlist `durvars' {
-		replace `var' = 0 if `var' < 0 & !mi(`var')
-		bysort `enum': egen `col_sub_dur' = mean(`var')
-		_updatesheet `col_sub_dur' using `duration_sheet', by(`enum') rename(`var')
-		drop `col_sub_dur'
+	if "`durvars'" != ""{
+		foreach var of varlist `durvars' {
+			replace `var' = 0 if `var' < 0 & !mi(`var')
+			bysort `enum': egen `col_sub_dur' = mean(`var')
+			_updatesheet `col_sub_dur' using `duration_sheet', by(`enum') rename(`var')
+			drop `col_sub_dur'
+		}
 	}
 	
 	// create summary sheet
 	preserve
 
 	// restrict to specified number of days
 	local today = date(c(current_date), "DMY")
-	keep if dofc(`subdate') > `today' - `days' 
-
+	sum `subdate'
+	//tests if the date is already in td format
+	if `r(max)' < 100000{
+		keep if `subdate' > `today' - `days'
+	}
+	//if not already in td format, uses dofc for date
+	else{
+		keep if dofc(`subdate') > `today' - `days'
+	}
+	
+	//create collapse command for calculating subtotals by enumerator
+	if `have_duration' == 0{

```

Comment:
I know it's a small stylistic thing but can you add spaces before your `{` in your if statements? i.e.

```Stata
if `have_duration' == 0 {
```